insights/parsers/lspci.py

```diff
@@ -69,14 +70,20 @@ def parse_content(self, content):
         # Parse kernel driver lines
         self.data = {}
         bus_device_function = ""
+        bus_device_function_re = re.compile(r'[0-9a-f]+:[0-9a-f]+.[0-9a-f]+')
+
+        fields = ["Subsystem", "Kernel driver in use", "Kernel modules"]
+
         for line in get_active_lines(content):
             parts = line.split(None)
-            if parts[0] and parts[0] not in ['Subsystem:', 'Kernel'] \
-                    and len(parts[0].split(':')) == 2 and len(parts[0].split('.')) == 2:
+            if not parts:
```

Comment:
@akshay196 Is this check required? because the `get_active_lines()` should handle the blank and the commented lines.

Reply:
Please see below diff altogether which should get 100% coverage

```diff
diff --git a/insights/parsers/lspci.py b/insights/parsers/lspci.py
index f54d9aaf6edf..bde7eff145eb 100644
--- a/insights/parsers/lspci.py
+++ b/insights/parsers/lspci.py
@@ -69,15 +69,13 @@ class LsPci(CommandParser, LogFileOutput):
             scanner(self)
         # Parse kernel driver lines
         self.data = {}
-        bus_device_function = ""
+        bus_device_function = None
         bus_device_function_re = re.compile(r'[0-9a-f]+:[0-9a-f]+.[0-9a-f]+')
 
         fields = ["Subsystem", "Kernel driver in use", "Kernel modules"]
 
         for line in get_active_lines(content):
             parts = line.split(None)
-            if not parts:
-                continue
 
             if parts[0] and bus_device_function_re.match(parts[0]):
                 bus_device_function = parts[0]
diff --git a/insights/parsers/tests/test_lspci.py b/insights/parsers/tests/test_lspci.py
index 7347b1c9b0f2..125b0d6cd2d7 100644
--- a/insights/parsers/tests/test_lspci.py
+++ b/insights/parsers/tests/test_lspci.py
@@ -205,7 +205,18 @@ LSPCI_DRIVER_DOC = """
 """.strip()
 
 
+LSPCI_OUTPUT_WITH_BAD_LINES = """
+03:00.0 Network controller: Intel Corporation Wireless 7260 (rev bb)
+        Subsystem: Intel Corporation Dual Band Wireless-AC 7260
+
+        Kernel driver in use: iwlwifi
+# a comment
+        Kernel modules: iwlwifi
+""".strip()
+
+
 def test_lspci():
+    LsPci.token_scan('centrino', 'Centrino')
     lspci_0 = LsPci(context_wrap(LSPCI_0))
     assert [i['raw_message'] for i in lspci_0.get("Intel Corporation")] == INTEL.splitlines()
     assert len(lspci_0.get("Network controller")) == 1
@@ -237,6 +248,10 @@ def test_lspci_driver():
     assert len(lspci_obj.data) == 0
     assert len(lspci_obj.pci_dev_list) == 0
 
+    output = LsPci(context_wrap(LSPCI_OUTPUT_WITH_BAD_LINES))
+    assert len(output.data) == 1
+    assert len(output.pci_dev_list) == 1
+
 
 def test_doc_examples():
     env = {
```

Reply:
Done.
@psachin Thanks for suggestions.